cmd/searcher/internal/search/search_grpc.go

```diff
@@ -38,3 +43,30 @@ func (s *Server) Search(req *proto.SearchRequest, stream proto.SearcherService_S
 		},
 	})
 }
+
+func (s *Server) CacheState(_ context.Context, req *proto.CacheStateRequest) (*proto.CacheStateResponse, error) {
+	repo := api.RepoName(req.Repo)
+	commit := api.CommitID(req.CommitOid)
+
+	// TODO hybrid search? We need to correctly represent that state of hybrid
+	// search here. It sets path (which we use as nil). We likely won't be
+	// able to fully do this, but we should atleast store somewhere so that
+	// CacheState returns something sensical (right now it will just always
+	// say MISSING)
+	var state proto.CacheState
+	fromState := s.Service.Store.CacheState(repo, commit, nil)
+	switch fromState {
+	case diskcache.StateMissing:
+		state = proto.CacheState_MISSING
+	case diskcache.StateFetching:
+		state = proto.CacheState_FETCHING
+	case diskcache.StateCached:
+		state = proto.CacheState_CACHED
+	default:
+		return nil, errors.Errorf("unknown diskcache.State %q", fromState)
```

Comment:
Best practice for gRPC is to return errors that include a gRPC status code. In this case, `Internal` seems appropriate since the error is not actionable by the client.

```suggestion
		return nil, status.Errorf(codes.Internal, "unknown diskcache.State %q", fromState)
```